### PR TITLE
GitHub MockApi for Testing (+ Client-Tests)

### LINKED ISSUE #103: Create Mock API for GitHub Actions Tests

### PR DESCRIPTION
**Pull request by kolioOtSofia**
…t be always working now)

Closes: #103

### CODE REVIEW

File: packages/AutoTDD-Tests.package/ATDDGitHubActionParserTest.class/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+An ATDDGitHubActionParserTest is class that tests the methods from the ATDDGitHubActionsParser whether they parse as they should to.
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionParserTest.class/README.md]:
```suggestion
Remove class comment
```
There are currently no class comments for test classes, let's keep it this way so we don't make ourselves responsible for even more :D

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsClientTest.class/instance/testApiConnect.st [hunk 2]
```diff
@@ -0,0 +1,10 @@
+as yet unclassified
+testApiConnect
+	
+	| client response |
+	
+	client := ATDDMockGitHubActionsClient new. 
+	response := client requestOn: ''. 
+	self 
+		assert: response equals: self defaultAnswer
+
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsClientTest.class/instance/testApiConnect.st]:
```suggestion
```
remove line at the end of file

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsClientTest.class/instance/testApiConnect.st [hunk 3]
```diff
@@ -0,0 +1,10 @@
+as yet unclassified
+testApiConnect
+	
+	| client response |
+	
+	client := ATDDMockGitHubActionsClient new. 
+	response := client requestOn: ''. 
+	self 
+		assert: response equals: self defaultAnswer
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsClientTest.class/instance/testApiConnect.st]:
```suggestion
	response := client requestOn: ''. 
	self assert: response equals: self defaultAnswer
```
Keep identation like this so long as there is no cascade

File: packages/AutoTDD-Tests.package/ATDDGitHubActionsClientTest.class/instance/testInitiliazation.st [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+	
+	| client |
+	
+	client := ATDDMockGitHubActionsClient new. 
+	self 
+		assert: client webClient notNil;
+		assert: client webClient userAgent equals: client userAgent; 
+		assert: client gitHubActionsApiUrl equals: 'https://api.github.com/' 
+		
+
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionsClientTest.class/instance/testInitiliazation.st]:
```suggestion
```
Remove empty lines

File: packages/AutoTDD-Tests.package/ATDDRemoteClientTests.class/instance/setUp.st [hunk 5]
```diff
@@ -0,0 +1,5 @@
+as yet unclassified
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDRemoteClientTests.class/instance/setUp.st]:
```suggestion
classify (for all methods in this class)
```

File: packages/AutoTDD-Tests.package/AutoTDDTestsServiceProvider.class/class/initialize.st [hunk 6]
```diff
@@ -0,0 +1,3 @@
+initialization
+initialize 
+	ServiceRegistry current buildProvider: self new
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/AutoTDDTestsServiceProvider.class/class/initialize.st]:
```suggestion
initialize

	ServiceRegistry current buildProvider: self new
```

**Review by SebastianWagner2 — CHANGES_REQUESTED** [review]

File: packages/AutoTDD-Tests.package/ATDDGitHubActionParserTest.class/README.md (hunk 1 above)
Comment (kolioOtSofia) [reply packages/AutoTDD-Tests.package/ATDDGitHubActionParserTest.class/README.md]: I will be glad to remove it hah

File: packages/AutoTDD-Tests.package/ATDDGitHubActionParserTest.class/instance/sampleNameJson.st [hunk 7]
```diff
@@ -0,0 +1,5 @@
+utility
```
Comment (benjamin-frost) [inline packages/AutoTDD-Tests.package/ATDDGitHubActionParserTest.class/instance/sampleNameJson.st:1]: `utility` probably does not suit well as a category here

**Review by benjamin-frost — CHANGES_REQUESTED** [review]

**Review by benjamin-frost — APPROVED** [review]

File: packages/AutoTDD-Core.package/ATDDGitHubActionsClient.class/instance/apiUrl..st [hunk 8]
```diff
@@ -0,0 +1,3 @@
+accessing
+apiUrl: aString
+	apiUrl := aString
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Core.package/ATDDGitHubActionsClient.class/instance/apiUrl..st]:
```suggestion
apiUrl: aString

	apiUrl := aString
```

File: packages/AutoTDD-Core.package/ATDDGitHubActionsClient.class/instance/apiUrl.st [hunk 9]
```diff
@@ -0,0 +1,3 @@
+accessing
+apiUrl
+	^ apiUrl ifNil: [ apiUrl := 'https://api.github.com/' ]
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Core.package/ATDDGitHubActionsClient.class/instance/apiUrl.st]:
```suggestion
apiUrl

	^ apiUrl ifNil: [ apiUrl := 'https://api.github.com/' ]
```

File: packages/AutoTDD-Tests.package/ATDDRemoteClientTests.class/instance/testThrowingExceptions.st [hunk 10]
```diff
@@ -0,0 +1,7 @@
+testing
+testThrowingExceptions
+
+	self should: [ remoteClient throwExceptionForResponseCode: 401 ] raise: ATDDConnectionError.
+	self should: [ remoteClient throwExceptionForResponseCode: 403 ] raise: ATDDAccessDeniedError.
+	self should: [ remoteClient throwExceptionForResponseCode: 404 ] raise: ATDDNotFoundError
+
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Tests.package/ATDDRemoteClientTests.class/instance/testThrowingExceptions.st]:
```suggestion
	self should: [ remoteClient throwExceptionForResponseCode: 404 ] raise: ATDDNotFoundError
```
Remove line at the end

**Review by SebastianWagner2 — CHANGES_REQUESTED** [review]

File: packages/AutoTDD-Core.package/ATDDGitHubActionsClient.class/instance/apiUrl.st (hunk 9 above)
Comment (SebastianWagner2) [reply packages/AutoTDD-Core.package/ATDDGitHubActionsClient.class/instance/apiUrl.st]: Still open

**Review by SebastianWagner2 — CHANGES_REQUESTED** [review]

**Review by SebastianWagner2 — APPROVED** [review]

**Pull request merged**